tests/core/test_hdaccount.py

```diff
@@ -0,0 +1,88 @@
+    ValidationError,
+)
+
+from eth_account import (
+    Account,
+)
+from eth_account.hdaccount import (
+    ETHEREUM_DEFAULT_PATH,
+)
+
+Account.enable_unaudited_hdwallet_features()
+
+
+@pytest.mark.parametrize("mnemonic,account_path,expected_address", [
+    # Ganache
+    # https://github.com/trufflesuite/ganache-core/blob/master/test/accounts.js
+    (
+        "into trim cross then helmet popular suit hammer cart shrug oval student",
+        ETHEREUM_DEFAULT_PATH,
+        "0x604a95C9165Bc95aE016a5299dd7d400dDDBEa9A",
+    ),
+    # Metamask
+    # https://github.com/MetaMask/eth-hd-keyring/blob/master/test/index.js
+    (
+        "finish oppose decorate face calm tragic certain desk hour urge dinosaur mango",
+        ETHEREUM_DEFAULT_PATH,
+        "0x1c96099350f13D558464eC79B9bE4445AA0eF579",
+    ),
+    (
+        "finish oppose decorate face calm tragic certain desk hour urge dinosaur mango",
+        "m/44'/60'/0'/0/1",  # 2nd account index in path
+        "0x1b00AeD43a693F3a957F9FeB5cC08AFA031E37a0",
+    ),
+])
+def test_account_derivation(mnemonic, account_path, expected_address):
+    a = Account.from_mnemonic(mnemonic, account_path=account_path)
+    assert a.address == expected_address
+
+
+def test_account_restore():
+    a1, mnemonic = Account.create_with_mnemonic(num_words=24, passphrase="TESTING")
+    a2 = Account.from_mnemonic(mnemonic, passphrase="TESTING")
+    assert a1.address == a2.address
+
+
+def test_incorrect_size():
+    with pytest.raises(ValidationError, match="Language not detected .*"):
+        Account.from_mnemonic("this is not a seed phrase")
+
+
+def test_malformed_seed():
+    with pytest.raises(ValidationError, match=".* not a valid BIP39 mnemonic phrase!"):
+        # Missing 12th word
+        Account.from_mnemonic("into trim cross then helmet popular suit hammer cart shrug oval")
+
+
+def test_incorrect_checksum():
+    with pytest.raises(ValidationError, match=".* not a valid BIP39 mnemonic phrase!"):
+        # Moved 12th word of valid phrase to be 1st
+        Account.from_mnemonic(
+            "student into trim cross then helmet popular suit hammer cart shrug oval"
+        )
+
+
+def test_incorrect_num_words():
+    with pytest.raises(ValidationError, match="Invalid choice for number of words.*"):
+        Account.create_with_mnemonic(num_words=11)
+
+
+def test_bad_account_path1():
+    with pytest.raises(ValidationError, match="Path is not valid.*"):
+        Account.from_mnemonic(
+            "finish oppose decorate face calm tragic certain desk hour urge dinosaur mango",
+            account_path='not an account path'
+        )
+
+
+def test_bad_account_path2():
+    with pytest.raises(ValidationError, match="Path.*is not valid.*"):
+        Account.create_with_mnemonic(account_path='m/not/an/account/path')
+
+
+def test_unknown_language():
+    with pytest.raises(ValidationError, match="Invalid language choice.*"):
+        Account.create_with_mnemonic(language="pig latin")
```

Comment:
For `create_with_mnemonic`, will you make sure that you have happy path testing for `create_with_mnemonic` like you have for `from_mnemonic`? So testing for a good account path (with hardened and unhardened indices), with passphrase, without passphrase, etc. And maybe this really belongs in the `test_account_restore` test, so we make sure addresses are derived/recovered consistently, but I can also see it standing on it's own.

Reply:
From this comment https://github.com/ethereum/eth-account/pull/87#discussion_r391272737 I think that "creating an account with a passphrase and restoring with a different one will fail" would be a good happy-path test case. (Showing that with the right passphrase, you get the right account, and with the wrong one you don't)

Reply:
Actually, `test_account_restore` already had the happy path, so I added the unhappy path via 051988f